src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -1621,6 +1650,13 @@ private bool ExactPointerInference(TypeWithAnnotations source, TypeWithAnnotatio
                      target.Type is FunctionPointerTypeSymbol { Signature: { ParameterCount: int targetParameterCount } targetSignature } &&
                      sourceParameterCount == targetParameterCount)
             {
+                if (sourceSignature.RefKind != targetSignature.RefKind
+                    || sourceSignature.ParameterRefKinds != targetSignature.ParameterRefKinds
```

Comment:
>!= [](start = 57, length = 2)

What does this actually do? It looks like we are using SequenceEqual for lower/upper bound inference.

Reply:
Fixed and added a test to cover.

---
In reply to: [562997374](https://github.com/dotnet/roslyn/pull/50249#discussion_r562997374) [](ancestors = 562997374)